tests/test_options.py

```diff
@@ -35,6 +41,61 @@ def get_converters_with_table_options():
             yield converter
 
 
+def test_accelerator_options():
+    # Check the default options
+    ao = AcceleratorOptions()
+    assert ao.num_threads == 4, "Wrong default num_threads"
+    assert ao.device == AcceleratorDevice.AUTO, "Wrong default device"
+
+    # Use API
+    ao2 = AcceleratorOptions(num_threads=2, device=AcceleratorDevice.MPS)
+    ao3 = AcceleratorOptions(num_threads=3, device=AcceleratorDevice.CUDA)
+    assert ao2.num_threads == 2
+    assert ao2.device == AcceleratorDevice.MPS
+    assert ao3.num_threads == 3
+    assert ao3.device == AcceleratorDevice.CUDA
+
+    # Use envvars (regular + alternative) and default values
+    os.environ["OMP_NUM_THREADS"] = "1"
+    ao.__init__()
```

Comment:
do not test the `__init__()` we should check the complete object initialization. (also see comment above, we should not have a custom `__init__`)

Reply:
It is not a check for the `__init__()` but an [in-place reloading](https://docs.pydantic.dev/latest/concepts/pydantic_settings/#in-place-reloading) that allows to re-evaluate the new values of the envvars.